[DOC-6519] Release notes for v23.1.0-alpha.9 (#16683)
cockroachdb/docs: _includes/releases/v23.1/v23.1.0-alpha.9.md

## v23.1.0-alpha.9

Release Date: April 4, 2023

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v23-1-0-alpha-9-backward-incompatible-changes">Backward-incompatible changes</h3>

- The output of the [`SHOW RANGES`](show-ranges.html) command for the `crdb_internal.ranges` and `crdb_internal.ranges_no_leases` tables has been updated, and the previous output is deprecated. To enable the new command output, set the `sql.show_ranges_deprecated_behavior.enabled` [cluster setting](cluster-settings.html) to `false`. The new output will become default in v23.2. [#99618][#99618]

<h3 id="v23-1-0-alpha-9-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>

> - The [`CREATE CHANGEFEED`](create-changefeed.html) now allows you to limit the nodes that can execute a changefeed by including a locality filter in the `WITH` clause of the `CREATE CHANGEFEED` command. A node can execute the changefeed only if it was started started with a matching [`--locality`](cockroach-start#locality) flag. Replace `{locality}` with a comma-separated list of key-value pairs. [#99935][#99935]

Reviewer: ```suggestion
- The [`CREATE CHANGEFEED`](create-changefeed.html) statement now allows you to limit the nodes that can execute a changefeed by including a locality filter in the `WITH` clause. A node can execute the changefeed only if it was started with a matching [`--locality`](cockroach-start#locality) flag. Replace `{locality}` with a comma-separated list of key-value pairs. [#99935][#99935]
```

<h3 id="v23-1-0-alpha-9-sql-language-changes">SQL language changes</h3>

- The new `prepared_statements_cache_size` [cluster setting](cluster-settings.html) helps to prevent [prepared statement](savepoint.html#savepoints-and-prepared-statements) leaks by automatically deallocating the least-recently-used prepared statements when the cache reaches a given size. [#99254][#99254]

<h3 id="v23-1-0-alpha-9-operational-changes">Operational changes</h3>

- The new `COCKROACH_DISABLE_NODE_AND_TENANT_METRIC_LABELS` [environment variable](cockroach-commands.html#environment-variables) allows you to suppress metrics from a cluster's [Prometheus endpoint](monitoring-and-alerting.html#the-prometheus-endpoint) if they conflict with labels that are applied by external tools that collect metrics from the endpoint. Set the environment variable to a comma-separated list of key-value pairs. [#99820][#99820]

<h3 id="v23-1-0-alpha-9-db-console-changes">DB Console changes</h3>

> - The [Index Details](ui-databases-page.html#index-details) now displays the list of most-frequently-used index fingerprints to all users, rather than only to admin users, because the page now queries a view rather than a system table directly. [#99485][#99485]

Reviewer: ```suggestion
- The [**Index Details**](ui-databases-page.html#index-details) section of the **Databases** page now displays the list of most-frequently-used index fingerprints to all users, rather than only to `admin` users, because the page now queries a view rather than a system table directly. [#99485][#99485]
```

> - When you search or filter within the [Statements page](ui-statements-page.html) or [Transactions Page](ui-transactions-page.html), if you interactively sort the results using a column that was not part of the original query, a warning displays if you are viewing only a subset of the results, along with a suggestion to update the original query. [#99795][#99795]

Reviewer: ```suggestion
- When you search or filter within the [**Statements** page](ui-statements-page.html) or [**Transactions** Page](ui-transactions-page.html), if you interactively sort the results using a column that was not part of the original query, a warning displays if you are viewing only a subset of the results, along with a suggestion to update the original query. [#99795][#99795]
```

<h3 id="v23-1-0-alpha-9-miscellaneous">Miscellaneous</h3>

- Several computed columns have been added to `crdb_internal.statement_statistics_persisted` and `crdb_internal.transaction_statistics_persisted` views and indexed in the system tables referenced by the views:

    - `execution_count`
    - `service_latency`
    - `cpu_sql_nanos`
    - `contention_time`
    - `total_estimated_execution_time`
    - `p99_latency`

    [#99417][#99417]

<h3 id="v23-1-0-alpha-9-bug-fixes">Bug fixes</h3>

> - Fixed pagination bugs when searching or filtering within the [Databases](ui-databases-page.html) or viewing the details of a database. [#99513][#99513]

Reviewer: ```suggestion
- Fixed pagination bugs when searching or filtering within the [**Databases**](ui-databases-page.html) page or viewing the details of a database. [#99513][#99513]
```

> - Fixed a rare bug introduced in v22.2.0 that could cause a node to crash with an `attempting to append refresh spans after the tracked timestamp has moved forward` error when querying the `pg_catalog` and `crdb_internal` virtual tables. If you are experiencing this bug, set the `sql.distsql.use_streamer.enabled` [cluster setting](cluster-settings.html) to `false` before upgrading a cluster to v23.1. [#99443][#99443]

Reviewer: We have dedicated pages for `pg_catalog` and `crdb_internal` - suggest linking to those pages.

- Fixed a bug that could erroneously cause multiple garbage-collection jobs to be created when executing a [`DROP SCHEMA ... CASCADE`](drop-schema.html) command, one job for each table and one for the cascaded `DROP` itself. [#99706][#99706]

> - Fixed a bug in the [Schema Insights](ui-schema-insights.html) and [Table Insights](ui-table-insights.html) pages that prevented a recommendation to drop an index from being executed if the index's name contained a space. [#100023][#100023]

Reviewer: I assume this is correct, but haven't checked whether those are new pages that appeared since the last time I pulled.

```suggestion
- Fixed a bug in the **Schema Insights** and **Table Insights** views of the [**Insights** page](ui-insights-page.html) that prevented a recommendation to drop an index from being executed if the index's name contained a space. [#100023][#100023]
```

- Fixed a rare bug that prevented the garbage-collection job for a [`TRUNCATE`](truncate.html) command from successfully finishing if the table descriptor had already been garbage-collected. The garbage-collection job now succeeds in this situation. [#100009][#100009]
- Fixed a rare bug that could cause a query of a virtual table in the [`crdb_internal`](crdb-internal.html) or [`pg_catalog`](pg-catalog.html) system catalog to hang indefinitely if the query returned an error. [#99969][#99969]
- Fixed a bug introduced prior to v21.2 that could cause the SQL gateway node to crash if you [created a view](create-view.html) with a circular or self-referencing dependencies. This situation no longer crashes the node, and a `cyclic view dependency for relation` error is now logged. [#100159][#100159]

> - Several rare bugs have been fixed that could cause corruption in the existing primary index when a rollback occurs concurrent to adding or removing a [column family](https://www.cockroachlabs.com/docs/stable/column-families.html). This could lead to subsequent unavailability of the table. [#100030][#100030]

Reviewer: ```suggestion
- Several rare bugs have been fixed that could cause corruption in the existing primary index when a rollback occurs concurrent to adding or removing a [column family](column-families.html). This could lead to subsequent unavailability of the table. [#100030][#100030]
```

> - Fixed a bug in the [Statements Detail](ui-statements-page.html) and [Transactions Details](ui-transactions-page.html) pages that could cause a node to crash with an out-of-memory (OOM) exception. [#99550][#99550]

Reviewer: I'm not sure if "Statement Details" is still the name of that section of the UI. Tried to look in `cockroach demo` and didn't see it. It would be best to double-check the correct title. It looks like "Statement Fingerprint" now, but I'm only guessing that this is where the bugfix was applied.

```suggestion
- Fixed a bug in [**Statement Details**](ui-statements-page.html) and [**Transaction Details**](ui-transactions-page.html) that could cause a node to crash with an out-of-memory (OOM) exception. [#99550][#99550]
```
